components/Buttons/src/MDCButton.m

```diff
@@ -705,6 +705,7 @@ - (void)mdc_setAdjustsFontForContentSizeCategory:(BOOL)adjusts {
 - (void)contentSizeCategoryDidChange:(NSNotification *)notification {
   UIFont *font = [UIFont mdc_preferredFontForMaterialTextStyle:MDCFontTextStyleButton];
   self.titleLabel.font = font;
+  [self sizeToFit];
```

Comment:
Changing the font of a button will cause the button to re-layout, but it won't change the bounds.  We need to explicitly re-size the button.
In the earlier Dynamic Type PR on buttons they 'worked' as the bounds of the sample buttons handled all of the font sizes, but this was definitely a bug.